ledger-live-desktop: init at 1.12.0

thedavidmeister opened the pull request:
###### Motivation for this change

Adds ledger live desktop, the app that manages ledger S/X/blue

https://www.ledger.com/

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

alexarice commented on pkgs/applications/altcoins/ledger-live-desktop/default.nix:
```diff
@@ -0,0 +1,40 @@
+    sha256 = "0b919ilvv3zi17fnzngnnmg28dxqlq0dvj3fdvb50kh3ibrhdcfz";
+  };
+
+  buildInputs = [ appimage-run];
```
space after `appimage-run`

alexarice commented on pkgs/applications/altcoins/ledger-live-desktop/default.nix:
```diff
@@ -0,0 +1,40 @@
+
+  unpackPhase = ":";
+
+  installPhase = ''
```
I'm not sure on this but would it be a good idea to run the preInstall and postInstall hooks so that people can override them

alexarice commented on pkgs/applications/altcoins/ledger-live-desktop/default.nix:
```diff
@@ -0,0 +1,40 @@
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Ledger Live (Desktop)";
```
The description shouldn't really be the name of the package. I found "The companion to your Ledger hardware wallet" on their website but that's more of a slogan so there is almost certainly something better.

thedavidmeister replied:
@alexarice this is the description pasted from their github repo

alexarice replied:
meta.description should:

    Be capitalized.
    Not start with the package name.
    Not have a period at the end.

This is what contributing.md says and it doesn't seem like a helpful description, even if it is the description on their github

alexarice commented on pkgs/applications/altcoins/ledger-live-desktop/default.nix:
```diff
@@ -0,0 +1,40 @@
+    description = "Ledger Live (Desktop)";
+    longDescription = ''
+      Ledger Live (Desktop)
+    '';
```
Not much point including a longDescription if it is no longer than the description

thedavidmeister replied:
@alexarice how's this?

alexarice replied:
I personally feel a longDescription is unnecessary

thedavidmeister replied:
@alexarice how's this?

alexarice replied:
I would still only include the first line really but I think this is ok

alexarice commented on pkgs/applications/altcoins/ledger-live-desktop/default.nix:
```diff
@@ -0,0 +1,40 @@
+    longDescription = ''
+      Ledger Live (Desktop)
+    '';
+    homepage = https://www.ledger.com/live;
```
quotes round urls

thedavidmeister replied:
@alexarice why i thought urls were first class?

alexarice replied:
They are first class. There is an ongoing discussion on whether to try to avoid their usage though I looked at it yesterday and it is quite ongoing so you could leave this if you want

bhipple commented on pkgs/applications/altcoins/ledger-live-desktop/default.nix:
```diff
@@ -0,0 +1,40 @@
+    sha256 = "0b919ilvv3zi17fnzngnnmg28dxqlq0dvj3fdvb50kh3ibrhdcfz";
+  };
+
+  buildInputs = [ appimage-run];
```
```suggestion
  buildInputs = [ appimage-run ];
```

thedavidmeister replied:
thanks

bhipple commented on pkgs/applications/altcoins/ledger-live-desktop/default.nix:
```diff
@@ -0,0 +1,40 @@
+}:
+
+stdenv.mkDerivation rec {
+  github-user = "LedgerHQ";
```
```suggestion
```

bhipple commented on pkgs/applications/altcoins/ledger-live-desktop/default.nix:
```diff
@@ -0,0 +1,40 @@
+  version = "1.9.1";
+
+  src = fetchurl {
+    url = "https://github.com/${github-user}/${pname}/releases/download/v${version}/${pname}-${version}-linux-x86_64.AppImage";
```
```suggestion
    url = "https://github.com/LedgerHQ/${pname}/releases/download/v${version}/${pname}-${version}-linux-x86_64.AppImage";
```

thedavidmeister replied:
@bhipple why would you do this?

alexarice replied:
I don't see how splitting the github-user to a separate variable makes anything neater

thedavidmeister replied:
@alexarice i like it, easier copypasta for the next app image i submit ;)

bhipple replied:
Possibly, but it's not a convention used elsewhere in NixPkgs, and there's some value in consistency given how huge the distribution is.  

Another consideration is that attributes at the top-level of the derivation, such as `buildInputs` or `pname` -- or your `github-user` -- make it into the hash and become part of the propagated description of the package in the derivation.  E.g., it'll show up in `nix show-derivation`, and if the github repo moves/changes names without changing src it'll change the hash and cause a rebuild.  

Not a very big deal and mostly a bike shed, but personally I'd still recommend removing it.

bhipple commented on pkgs/applications/altcoins/ledger-live-desktop/default.nix:
```diff
@@ -0,0 +1,40 @@
+  pname = "ledger-live-desktop";
+  version = "1.9.1";
+
+  src = fetchurl {
```
Consider using `fetchFromGitHub`, which is more efficient because it fetches just the tar.gz of the given commit (shallow), instead of doing a full git clone and throwing away the `.git` dir.

thedavidmeister replied:
@bhipple but we don't want the tarball of the git repo we want the app image that is built and shipped as part of the releases

thedavidmeister commented on pkgs/applications/altcoins/ledger-live-desktop/default.nix:
```diff
@@ -0,0 +1,42 @@
+  buildInputs = [ appimage-run ];
+
+  unpackPhase = ":";
+  preInstall = ":";
```
@alexarice like this?

alexarice replied:
like https://github.com/NixOS/nixpkgs/blob/8f5edef47e2d7d3667148d7ea8122fb4ec6794d1/pkgs/applications/editors/sublime/3/common.nix#L86

thedavidmeister commented on pkgs/applications/altcoins/ledger-live-desktop/default.nix:
```diff
@@ -0,0 +1,44 @@
+  unpackPhase = ":";
+
+  installPhase = ''
+    runHook preInstall
```
@alexarice hooks right now?

alexarice replied:
looks good

alexarice commented:
You will need to squash the commits into one adding yourself as a maintainer and one adding the package at some point

thedavidmeister committed:
ledger-live-desktop: init at 1.9.1

Update pkgs/applications/altcoins/ledger-live-desktop/default.nix

Co-Authored-By: Benjamin Hipple <bhipple@protonmail.com>
add preInstall and postInstall for ledger-live-desktop

updated longDescription for ledger-live-desktop

resolving feedback

thedavidmeister commented:
@alexarice squashed, we good?

alexarice commented:
I have already approved and cannot do anything else

yegortimoshenko commented:
@GrahamcOfBorg build ledger-live-desktop

lukateras committed:
ledger-live-desktop: add desktop item, address review

lukateras committed:
ledger-live-desktop: 1.9.1 -> 1.12.0

lukateras reviewed (commented):
I'm going to push a commit that addresses this review, please check and tell if you're comfortable with it. Note that it also adds an XDG desktop item.

yegortimoshenko commented:
@GrahamcOfBorg build ledger-live-desktop

thedavidmeister commented:
@yegortimoshenko i'm happy with your changes, let's merge :)
